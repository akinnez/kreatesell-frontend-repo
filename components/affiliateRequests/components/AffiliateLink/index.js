import { useRef } from "react";
import { Typography, Button } from "antd";
import styles from "./index.module.scss";
import { showToast } from "utils";

const { Text } = Typography;

const AffiliateLink = () => {
  const linkElement = useRef();

  const handleCopy = async () => {
    try {
      const link = linkElement.current.textContent;

      if ("clipboard" in navigator) {
        await navigator.clipboard.writeText(link);
        showToast("Affiliate link copied", "success");
      } else {
        showToast("Unable to copy affiliate link in this browser", "info");
      }
    } catch (err) {
      showToast("Something went wrong. Can not copy link at this time", "warn");
    }
  };

  return (
    <section className={styles.link__container}>
      <div className={styles.label}>
        <Text>Affiliate Link</Text>
      </div>
      <div className={styles.link}>
        <span ref={linkElement}>
          23dghy5668jjkkloo//pred892gykk//ujjikhs7778
        </span>
      </div>
      <Button className={styles.link__btn} onClick={handleCopy}>
        Copy Link
      </Button>
    </section>
  );
};

export default AffiliateLink;
