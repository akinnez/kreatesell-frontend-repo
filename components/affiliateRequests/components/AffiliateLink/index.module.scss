@use "public/css/colors";
@use "public/css/mixins";

.label {
  margin-bottom: 10px;

  & > span {
    font-weight: 500;
    font-size: 16px;
    color: colors.$black-100;
  }
}

.link__container {
  max-width: 790px;
  @include mixins.flex(row, flex-start, center);
  flex-wrap: wrap;
  gap: 15px;
}

.link {
  height: 45px;
  width: 100%;
  max-width: 670px;
  padding: 10px 15px;
  border: 1px dashed colors.$base-blue;
  border-radius: 8px;
  background-color: colors.$base-green-300;
  white-space: nowrap;
  overflow-x: hidden;
  text-overflow: ellipsis;

  & > span {
    color: colors.$black-100;
    font-size: 18px;
    line-height: 1;
  }
}

.link__btn {
  height: 45px;
  padding: 10px 15px;
  border-radius: 8px;
  border: 1px solid colors.$base-blue;
  color: colors.$base-blue;
  font-weight: bold;
  font-size: 12px;
  line-height: 1;
}
