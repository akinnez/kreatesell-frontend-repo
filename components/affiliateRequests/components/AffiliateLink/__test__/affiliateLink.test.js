import AffiliateLink from "..";
import renderer from "react-test-renderer";

it("renders a snapshot of <AffiliateLink />", () => {
  const link = renderer.create(<AffiliateLink />).toJSON();
  expect(link).toMatchSnapshot();
});
