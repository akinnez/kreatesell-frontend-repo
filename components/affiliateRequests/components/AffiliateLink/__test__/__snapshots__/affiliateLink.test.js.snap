// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a snapshot of <AffiliateLink /> 1`] = `
<section
  className="link__container"
>
  <div
    className="label"
  >
    <span
      className="ant-typography"
      onClick={null}
      style={
        Object {
          "WebkitLineClamp": undefined,
        }
      }
    >
      Affiliate Link
    </span>
  </div>
  <div
    className="link"
  >
    <span>
      23dghy5668jjkkloo//pred892gykk//ujjikhs7778
    </span>
  </div>
  <button
    className="ant-btn link__btn"
    onClick={[Function]}
    type="button"
  >
    <span>
      Copy Link
    </span>
  </button>
</section>
`;
